java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -213,35 +211,49 @@ private void asyncOpenSelectedProjects(CompletableFuture f, List<FileObject> pro
             try {
-                JavaSource.create(ClasspathInfo.create(ClassPath.EMPTY, ClassPath.EMPTY, ClassPath.EMPTY))
-                          .runWhenScanFinished(cc -> {
-                  if (client.getNbCodeCapabilities().hasStatusBarMessageSupport()) {
-                        client.showStatusBarMessage(new ShowStatusMessageParams(MessageType.Info, INDEXING_COMPLETED, 0));
-                  } else {
-                        client.showMessage(new ShowStatusMessageParams(MessageType.Info, INDEXING_COMPLETED, 0));
-                  }
-                  //todo: refresh diagnostics all open editor?
-                }, true);
+                final ClasspathInfo info = ClasspathInfo.create(ClassPath.EMPTY, ClassPath.EMPTY, ClassPath.EMPTY);
+                final JavaSource source = JavaSource.create(info);
+                if (source == null) {
+                    SERVER_INIT_RP.post(() -> {
+                        final String msg = NO_JAVA_SUPPORT + System.getProperty("java.version");
+                        showStatusBarMessage(MessageType.Error, msg, 5000);
+                    });
+                } else {
+                    source.runWhenScanFinished(cc -> {
+                        showStatusBarMessage(MessageType.Info, INDEXING_COMPLETED, 0);
+                    }, true);
+                }
+                return source;
             } catch (IOException ex) {
                 throw new IllegalStateException(ex);
             }
         }
+
+        private void showStatusBarMessage(final MessageType type, final String msg, int timeout) {
+            if (client.getNbCodeCapabilities().hasStatusBarMessageSupport()) {
+                client.showStatusBarMessage(new ShowStatusMessageParams(type, msg, timeout));
+            } else {
+                client.showMessage(new ShowStatusMessageParams(type, msg, timeout));
+            }
+        }
         
-        private InitializeResult constructInitResponse() {
+        private InitializeResult constructInitResponse(JavaSource src) {
             ServerCapabilities capabilities = new ServerCapabilities();
-            capabilities.setTextDocumentSync(TextDocumentSyncKind.Incremental);
-            CompletionOptions completionOptions = new CompletionOptions();
-            completionOptions.setResolveProvider(true);
-            completionOptions.setTriggerCharacters(Collections.singletonList("."));
-            capabilities.setCompletionProvider(completionOptions);
-            capabilities.setCodeActionProvider(true);
-            capabilities.setDocumentSymbolProvider(true);
-            capabilities.setDefinitionProvider(true);
-            capabilities.setDocumentHighlightProvider(true);
-            capabilities.setReferencesProvider(true);
-            capabilities.setExecuteCommandProvider(new ExecuteCommandOptions(Arrays.asList(JAVA_BUILD_WORKSPACE, GRAALVM_PAUSE_SCRIPT)));
+            if (src != null) {
```

Comment:
If there is no `JavaSource` (e.g. no Java infrastructure installed), then our server is started, but it has no code editing capabilities - e.g. no additional queries needed `JavaSource` shall arrive.